src/ast/passes/resolve_imports.cpp

```diff
@@ -0,0 +1,54 @@
+#include <iostream>
+#include <sstream>
+#include <unordered_set>
+
+#include "ast/passes/resolve_imports.h"
+#include "ast/visitor.h"
+#include "bpftrace.h"
+#include "log.h"
+
+namespace bpftrace::ast {
+
+// Load imports emit unprocessed source, objects, etc.
+class ResolveImports : public Visitor<ResolveImports> {
+public:
+  ResolveImports(BPFtrace &bpftrace, const std::vector<std::string> &paths)
+      : bpftrace_(bpftrace), paths_(paths) {};
+
+  using Visitor<ResolveImports>::visit;
+  void visit(Import &imp);
+
+  // Should be called after the visit, this will move the result.
+  Imports imports()
+  {
+    return std::move(imports_);
+  }
+
+private:
+  BPFtrace &bpftrace_;
+  const std::vector<std::string> &paths_;
+  Imports imports_;
+};
+
+void ResolveImports::visit(Import &imp)
+{
+  if (!bpftrace_.config_->get(ConfigKeyBool::unstable_import)) {
```

Comment:
I'm going to make a helper for this sort of thing that passes can call.

Reply:
Sure, I think it's fine though. I'd still want to check it within the context of the pass, although maybe you're talking about just standardizing the message? That sounds fine.